### PR TITLE
Adds recognition of the 'crunch' R5RS subset

### PR DESCRIPTION
**Pull request by dleslie**
Crunch is a subset of R5RS that the crunch egg can heavily optimize via
c++ compilation. This change allows geiser to report to chicken
programmers whether the function is found within that subset, easing
development.

Details on the crunch egg can be found at:
http://wiki.call-cc.org/eggref/4/crunch

### CODE REVIEW

**Comment by dleslie** [comment]
This is useful for just about anyone who is interested in writing performance-aware chicken code

File: scheme/chicken/geiser/emacs.scm [hunk 1]
```diff
@@ -172,6 +172,10 @@
...
...r subs16vector subs32vector subs8vector subu16vector subu32vector subu8vector u16vector-length u16vector-ref u16vector-set! u32vector-length u32vector-ref u32vector-set! u8vector-length u8vector-ref u8vector-set! blob->f32vector blob->f32vector/shared blob->f64vector blob->f64vector/shared blob->s16vector blob->s16vector/shared blob->s32vector blob->s32vector/shared blob->s8vector blob->s8vector/shared blob->string blob->string/shared blob->u16vector blob->u16vector/shared blob->u32vector blob->u32vector/shared blob->u8vector blob->u8vector/shared f32vector->blob f32vector->blob/shared f64vector->blob f64vector->blob/shared s16vector->blob s16vector->blob/shared s32vector->blob s32vector->blob/shared s8vector->blob s8vector->blob/shared string->blob string->blob/shared u16vector->blob u16vector->blob/shared u32vector->blob u32vector->blob/shared u8vector->blob u8vector->blob/shared flush-output void error exit argc argv-ref pointer-u8-ref pointer-s8-ref pointer-u16-ref pointer-s16-ref pointer-u32-ref pointer-s32-ref pointer-f32-ref pointer-f64-ref pointer-u8-set! pointer-s8-set! pointer-u16-set! pointer-s16-set! pointer-u32-set! pointer-s32-set! pointer-f32-set! pointer-f64-set!)))
```
Comment (jaor) [inline scheme/chicken/geiser/emacs.scm]: could we break down this huge line into separate 80-cols lines? what about sorting the list elements? it'd make perusing the list easier...

**Comment by dleslie** [comment]
Done

**Comment by jaor** [comment]
thanks, dan.  merged!

**Pull request merged**